dt-assets/js/modular-list.js

```diff
@@ -32,70 +33,96 @@
+      return cachedFilter;
+    }
+    return { query: {} };
   }
 
-  //set up custom cached filter
-  // a valid filter ID in the query params will skip to clicking the relevant filter.
-  if ( !isInFilterList(filterID) && query_param_custom_filter && !window.lodash.isEmpty(query_param_custom_filter) && query_param_custom_filter.type === "custom_filter" ){
-    query_param_custom_filter.query.offset = 0;
-    add_custom_filter(query_param_custom_filter.name, "default", query_param_custom_filter.query, query_param_custom_filter.labels, false)
-  } else if ( !isInFilterList(filterID) && cached_filter && !window.lodash.isEmpty(cached_filter) && cached_filter.type === "custom_filter" ) {
-    cached_filter.query.offset = 0;
-    add_custom_filter(cached_filter.name, "default", cached_filter.query, cached_filter.labels, false)
-  } else {
-    //check select filter
-    if ( current_filter.ID ){
-      //open the filter tabs
-      $(`#list-filter-tabs [data-id='${window.lodash.escape( current_filter.tab )}'] a`).click()
-      let filter_element = $(`input[name=view][data-id="${window.lodash.escape( current_filter.ID )}"].js-list-view`)
-      if ( filter_element.length ){
-        filter_element.prop('checked', true);
+  function setup_custom_cached_filter(urlCustomFilter, cachedFilter, currentFilter) {
+    const { filterID } = get_url_query_params()
+
+    if (!isInFilterList(filterID) && urlCustomFilter && !window.lodash.isEmpty(urlCustomFilter) && urlCustomFilter.type === "custom_filter") {
+      urlCustomFilter.query.offset = 0;
+      add_custom_filter(urlCustomFilter.name, "default", urlCustomFilter.query, urlCustomFilter.labels, false);
+    } else if (!isInFilterList(filterID) && cachedFilter && !window.lodash.isEmpty(cachedFilter) && cachedFilter.type === "custom_filter") {
+      cachedFilter.query.offset = 0;
+      add_custom_filter(cachedFilter.name, "default", cachedFilter.query, cachedFilter.labels, false);
+    } else {
+      //check select filter
+      if (currentFilter.ID) {
+        //open the filter tabs
+        $(`#list-filter-tabs [data-id='${window.lodash.escape(currentFilter.tab)}'] a`).click();
+        let filter_element = $(`input[name=view][data-id="${window.lodash.escape(currentFilter.ID)}"].js-list-view`);
+        if (filter_element.length) {
+          filter_element.prop('checked', true);
+        } else {
+          check_first_filter();
+        }
       } else {
-        check_first_filter()
+        check_first_filter();
       }
-    } else {
-      check_first_filter()
     }
   }
 
-  //determine list columns
-  if ( window.lodash.isEmpty(fields_to_show_in_table)){
-    fields_to_show_in_table = list_settings.fields_to_show_in_table
+  let check_first_filter = function (){
+    $('#list-filter-tabs .accordion-item a')[0].click()
+    $($('.js-list-view')[0]).prop('checked', true)
+  }
+
+  function determineListColumns(fieldsToShowInTable, listSettings) {
```

Comment:
@squigglybob why pass in `listSettings` when it is available in root?

Reply:
that's a good question. I was trying to go for the paradigm of having every dependency passed into the function through the params. But not sure if it's really necessary or not.

I was also trying to have the functions have no side effects, like modifying incoming params or global params, but failed in that also.

I'm happy to cut them out if you would prefer to just grab the global?